std/string.d

```diff
@@ -4476,33 +4476,21 @@ bool isNumeric(const(char)[] s, in bool bAllowSep = false) @safe pure
  *  but this one is the standard one.
  */
 
-char[] soundex(const(char)[] string, char[] buffer = null)
-    @safe pure nothrow
-in
-{
-    assert(!buffer.ptr || buffer.length >= 4);
-}
-out (result)
-{
-    if (result.ptr)
-    {
-        assert(result.length == 4);
-        assert(result[0] >= 'A' && result[0] <= 'Z');
-        foreach (char c; result[1 .. 4])
-            assert(c >= '0' && c <= '6');
-    }
-}
-body
+char[4] soundexer(Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range) ||
+        isSomeString!Range)
 {
+    alias C = Unqual!(ElementEncodingType!Range);
+
     static immutable dex =
         // ABCDEFGHIJKLMNOPQRSTUVWXYZ
-        "01230120022455012623010202";
-
-    int b = 0;
-    char lastc;
-    foreach (char cs; string)
-    {   auto c = cs;        // necessary because cs is final
+          "01230120022455012623010202";
 
+    char[4] result;
+    size_t b = 0;
+    C lastc;
+    foreach (C c; str)
+    {
         if (c >= 'a' && c <= 'z')
             c -= 'a' - 'A';
         else if (c >= 'A' && c <= 'Z')
```

Comment:
I believe that should be note that this only works for english. The foreach + the char math will result in some weird stuff.